src/switch/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { DNode } from '@dojo/framework/core/interfaces';
+import focus from '@dojo/framework/core/middleware/focus';
+import coreTheme from '@dojo/framework/core/middleware/theme';
+import { uuid } from '@dojo/framework/core/util';
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+import { formatAriaProperties } from '../common/util';
+import Label from '../label';
+import * as css from '../theme/default/switch.m.css';
+
+interface SwitchProperties {
+	/** Custom aria attributes */
+	aria?: { [key: string]: string | null };
+	/**  Checked/unchecked property of the switch */
+	checked?: boolean;
+	/** Whether the switch is disabled or clickable */
+	disabled?: boolean;
+	/** The label to be displayed for the switch */
+	label?: string;
+	/** Whether the label is hidden or displayed */
+	labelHidden?: boolean;
+	/** The name attribute for the switch */
+	name?: string;
+	/** Label to show in the "off" position of the switch */
+	offLabel?: DNode;
+	/** Handler for events triggered by switch losing focus */
+	onBlur?(): void;
+	/** Handler for events triggered by "on focus" */
+	onFocus?(): void;
+	/** Label to show in the "on" position of the switch */
+	onLabel?: DNode;
+	/** Handler for events triggered by "on out" */
+	onOut?(): void;
+	/** Handler for events triggered by "on over" */
+	onOver?(): void;
+	/** Handler for when the value of the widget changes */
+	onValue?(checked: boolean): void;
+	/** Makes the switch readonly (it may be focused but not changed) */
+	readOnly?: boolean;
+	/** Determines if this input is required, styles accordingly */
+	required?: boolean;
+	/** Determines if this input is valid */
+	valid?: boolean;
+	/** The current value */
+	value?: string;
+	/** `id` set on the root button DOM node */
+	widgetId?: string;
+}
+
+const factory = create({ coreTheme, focus }).properties<SwitchProperties>();
+
+export default factory(function Switch({ properties, middleware: { coreTheme, focus } }) {
+	const {
+		aria = {},
+		checked = false,
+		classes,
+		disabled,
+		label,
+		labelHidden,
+		name,
+		offLabel,
+		onBlur,
+		onFocus,
+		onValue,
+		onLabel,
+		onOut,
+		onOver,
+		readOnly,
+		required,
+		theme,
+		valid,
+		value,
+		widgetId = uuid()
```

Comment:
if you remove `widgetId` as a property and remove `uuid`, you can use `id` passed into the factory function as per https://github.com/dojo/widgets/blob/master/src/menu/index.tsx#L289